homeassistant/components/obihai/sensor.py

```diff
@@ -47,33 +47,26 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     pyobihai = PyObihai()
 
     login = pyobihai.check_account(host, username, password)
-    if login:
+    if not login:
+        _LOGGER.error("Invalid credentials")
+        return False
```

Comment:
Just `return`. Nothing is checking this return value.

Reply:
Done! Thanks for all your help :)